### PR TITLE
[fix] Small fixes

### PR DESCRIPTION
**Pull request by yoannmoinet**
### What and why?

https://github.com/DataDog/build-plugins/pull/112 introduced a bug in the sender of our telemetry plugin.

### How?

- Transform set into array right before sending metrics to DD.
- Add some comments here and there.
- Narrow down the entry filepath detection on xspack's build-report.

### CODE REVIEW

File: packages/plugins/build-report/src/xpack.ts [hunk 1]
```diff
@@ -367,7 +367,9 @@ export const getXpackPlugin =
                     // Filter the ones that includes the entry name.
                     .filter(
                         (f) => f.includes(name) || (entrypoint.name && f.includes(entrypoint.name)),
-                    )[0];
+                    )
+                    // Only keep JS files.
+                    .filter((f) => getType(f) === 'js')[0];
```
Comment (Ayc0) [inline packages/plugins/build-report/src/xpack.ts:372]: To note: if you use `filter` to then return the 1st element, you may want to use `.find` instead

File: packages/plugins/build-report/src/helpers.ts [hunk 2]
```diff
@@ -22,6 +22,7 @@ import path from 'path';
 const EXTENSION_RX = /\.(?!.*(?:\.|\/|\\))(\w{1,})/g;
 
 // Will match any type of query characters.
+// ? or %3F or |.
```
Comment (Ayc0) [inline packages/plugins/build-report/src/helpers.ts:25]: what is `%3F`?

Also you should remove the `.` from this comment as I first understood that as "will match `|.`"

**Review by Ayc0 — APPROVED** [review]

File: packages/plugins/build-report/src/xpack.ts (hunk 1 above)
Comment (yoannmoinet) [reply packages/plugins/build-report/src/xpack.ts:372]: Fair

File: packages/plugins/build-report/src/helpers.ts (hunk 2 above)
Comment (yoannmoinet) [reply packages/plugins/build-report/src/helpers.ts:25]: `%3F` is `?` url encoded.

File: packages/plugins/build-report/src/xpack.ts (hunk 1 above)
Comment (Ayc0) [reply packages/plugins/build-report/src/xpack.ts:372]: This will avoid creating a new array

**Pull request merged**